[Core Keeper] Added Steam Datagram Relay (SDR) support and world savi…

K3lvin4SY opened the pull request:
# Description

<!-- Please explain what is being changed or added as a short overview for this PR. Also, link existing relevant issues if they exist with resolves # -->

- Added option to use Steam Datagram Relay (SDR) instead of a port. Can be used to ignore port issues.
Example: https://github.com/pelican-eggs/eggs/issues/2525
This solution doesn't fix the opened issue, but it's a workaround to make it work.

- Added functionality for saving the world when stopping the server. Previously it would kill of the server without letting it save the world first. This would result in some progress being lost. 

## Checklist for all submissions

<!-- insert X into the brackets to mark it as done (i.e. [x]). You can click preview to make the links appear clickable. -->

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you tested and reviewed your changes with confidence that everything works?
* [X] Did you branch your changes and PR from that branch and not from your master branch?

<!-- If this is an egg update fill these out -->

* [X] You verify that the start command applied does not use a shell script
  * [X] If some script is needed then it is part of a current yolk or a PR to add one
* [X] The egg was exported from the panel

QuintenQVD0 commented:
You are messing things up. 

You can not change the email of the egg
And you exported a ptero exported egg in the pelican one. Please adjust.

QuintenQVD0 commented:
And please shorten the variable description.

K3lvin4SY commented:
Thanks for pointing that out! I've made the adjustments and fixed the issues:

- Reverted the email change for the egg.
- Corrected the Pelican egg.
- Shortened the variable description as requested.
Please let me know if there's anything else.

Thanks! =)

ImKringle commented:
No longer runs on the latest version (Just figured id mention, ran into this debugging my own)
```
Caught fatal signal - signo:11 code:1 errno:0 addr:0x20
Obtained 24 stack frames.
#0  0x007f5296ece050 in __sigaction
#1  0x007f520fa1cee0 in BumblelionLogger::TryToLoadConfigurationFile()
#2  0x007f520fa1ccd8 in BumblelionLogger::BumblelionLogger()
#3  0x007f520fa1d870 in BumblelionLogger::GetInstance()
#4  0x007f520fa1cbf6 in DbgLogAreaFlags_FnInOut
#5  0x007f520fa34d79 in OutstandingPlayFabRequestManager::OutstandingRequest::OutstandingRequest()
#6  0x007f520fa34bfd in OutstandingPlayFabRequestManager::OutstandingRequestList::OutstandingRequestList()
#7  0x007f520fa3a6ab in _GLOBAL__sub_I_PlayFabServiceManagerImpl.cpp
#8  0x007f52991f89ce in __nptl_change_stack_perm
#9  0x007f52991f8ab4 in __nptl_change_stack_perm
#10 0x007f5296fe1024 in _dl_catch_exception
#11 0x007f52991ff21e in _dl_rtld_di_serinfo
#12 0x007f5296fe0fca in _dl_catch_exception
#13 0x007f52991ff5b8 in _dl_find_dso_for_object
#14 0x007f5296f17438 in dlmopen
#15 0x007f5296fe0fca in _dl_catch_exception
#16 0x007f5296fe107f in _dl_catch_error
#17 0x007f5296f16f27 in dlerror
#18 0x007f5296f174e9 in dlopen
#19 0x007f5297fecf8e in PlayerMain(int, char**)
#20 0x007f5297fdd77d in PlayerMain(int, char**)
#21 0x007f5296eb924a in __libc_init_first
#22 0x007f5296eb9305 in __libc_start_main
#23 0x00000000201029 in (Unknown)
Segmentation fault (core dumped)
```
@K3lvin4SY

K3lvin4SY commented:
> No longer runs on the latest version (Just figured id mention, ran into this debugging my own)
> 
> ```
> Caught fatal signal - signo:11 code:1 errno:0 addr:0x20
> Obtained 24 stack frames.
> #0  0x007f5296ece050 in __sigaction
> #1  0x007f520fa1cee0 in BumblelionLogger::TryToLoadConfigurationFile()
> #2  0x007f520fa1ccd8 in BumblelionLogger::BumblelionLogger()
> #3  0x007f520fa1d870 in BumblelionLogger::GetInstance()
> #4  0x007f520fa1cbf6 in DbgLogAreaFlags_FnInOut
> #5  0x007f520fa34d79 in OutstandingPlayFabRequestManager::OutstandingRequest::OutstandingRequest()
> #6  0x007f520fa34bfd in OutstandingPlayFabRequestManager::OutstandingRequestList::OutstandingRequestList()
> #7  0x007f520fa3a6ab in _GLOBAL__sub_I_PlayFabServiceManagerImpl.cpp
> #8  0x007f52991f89ce in __nptl_change_stack_perm
> #9  0x007f52991f8ab4 in __nptl_change_stack_perm
> #10 0x007f5296fe1024 in _dl_catch_exception
> #11 0x007f52991ff21e in _dl_rtld_di_serinfo
> #12 0x007f5296fe0fca in _dl_catch_exception
> #13 0x007f52991ff5b8 in _dl_find_dso_for_object
> #14 0x007f5296f17438 in dlmopen
> #15 0x007f5296fe0fca in _dl_catch_exception
> #16 0x007f5296fe107f in _dl_catch_error
> #17 0x007f5296f16f27 in dlerror
> #18 0x007f5296f174e9 in dlopen
> #19 0x007f5297fecf8e in PlayerMain(int, char**)
> #20 0x007f5297fdd77d in PlayerMain(int, char**)
> #21 0x007f5296eb924a in __libc_init_first
> #22 0x007f5296eb9305 in __libc_start_main
> #23 0x00000000201029 in (Unknown)
> Segmentation fault (core dumped)
> ```
> 
> @K3lvin4SY

With the egg I've updated in #62 , the newest version works perfectly fine for me.

ImKringle commented:
> > No longer runs on the latest version (Just figured id mention, ran into this debugging my own)

> With the egg I've updated in #62 , the latest version works perfectly fine for me.

Fresh import and server creation using the egg from this PR, does not start:

Shutdown handler: initialize.
Preloaded 'libEOSSDK-Linux-Shipping.so'
Preloaded 'libPlayFabMultiplayer.so'
Preloaded 'lib_burst_generated.so'
Caught fatal signal - signo:11 code:1 errno:0 addr:0x20
Obtained 24 stack frames.
#0  0x007fd5e53ed050 in __sigaction
#1  0x007fd55c6fdee0 in BumblelionLogger::TryToLoadConfigurationFile()
#2  0x007fd55c6fdcd8 in BumblelionLogger::BumblelionLogger()
#3  0x007fd55c6fe870 in BumblelionLogger::GetInstance()
#4  0x007fd55c6fdbf6 in DbgLogAreaFlags_FnInOut
#5  0x007fd55c715d79 in OutstandingPlayFabRequestManager::OutstandingRequest::OutstandingRequest()
#6  0x007fd55c715bfd in OutstandingPlayFabRequestManager::OutstandingRequestList::OutstandingRequestList()
#7  0x007fd55c71b6ab in _GLOBAL__sub_I_PlayFabServiceManagerImpl.cpp
#8  0x007fd5e77179ce in __nptl_change_stack_perm
#9  0x007fd5e7717ab4 in __nptl_change_stack_perm
#10 0x007fd5e5500024 in _dl_catch_exception
#11 0x007fd5e771e21e in _dl_rtld_di_serinfo
#12 0x007fd5e54fffca in _dl_catch_exception
#13 0x007fd5e771e5b8 in _dl_find_dso_for_object
#14 0x007fd5e5436438 in dlmopen
#15 0x007fd5e54fffca in _dl_catch_exception
#16 0x007fd5e550007f in _dl_catch_error
#17 0x007fd5e5435f27 in dlerror
#18 0x007fd5e54364e9 in dlopen
#19 0x007fd5e650bf8e in PlayerMain(int, char**)
#20 0x007fd5e64fc77d in PlayerMain(int, char**)
#21 0x007fd5e53d824a in __libc_init_first
#22 0x007fd5e53d8305 in __libc_start_main
#23 0x00000000201029 in (Unknown)
/entrypoint.sh: line 89:    49 Segmentation fault      (core dumped) ./CoreKeeperServer -batchmode -logfile CoreKeeperServerLog.txt -world ${WORLD_INDEX} -worldname "${WORLD_NAME}" -worldseed ${WORLD_SEED} $([[ "${GAME_ID}" != "" ]] && echo -n " -gameid ${GAME_ID}") -maxplayers ${MAX_PLAYERS} -worldmode ${WORLD_MODE} $([[ ${STEAMRELAY_PORT} == 0 ]] && echo -n " -port ${SERVER_PORT}")
[3]+  Terminated              tail -f CoreKeeperServerLog.txt | grep -e '^\[userid:[0-9]*\] player' -e '^[^\[]'
container@pterodactyl~ Server marked as offline...
[Pterodactyl Daemon]: Aborting automatic restart, crash detection is disabled for this instance.

K3lvin4SY commented:
Okay thanks @ImKringle 
I can't seem to replicate the error you get.
Try to change the startup command with the following:

`export DISPLAY=:0; set -m; rm ./GameID.txt ./CoreKeeperServerLog.txt; touch ./CoreKeeperServerLog.txt; xvfb-run -s "-screen 0 {{DISPLAY_WIDTH}}x{{DISPLAY_HEIGHT}}x{{DISPLAY_DEPTH}} -ac -nolisten tcp -nolisten unix" ./CoreKeeperServer -batchmode -logfile CoreKeeperServerLog.txt -world {{WORLD_INDEX}} -worldname "{{WORLD_NAME}}" -worldseed {{WORLD_SEED}} $([[ "{{GAME_ID}}" != "" ]] && echo -n " -gameid {{GAME_ID}}") -maxplayers {{MAX_PLAYERS}} -worldmode {{WORLD_MODE}} $([[ {{STEAMRELAY_PORT}} == 0 ]] && echo -n " -port {{SERVER_PORT}}") & CKPID=$!; tail -f CoreKeeperServerLog.txt | grep -e '^\[userid:[0-9]*\] player' -e '^[^\[]' & LOGPID=$!; trap 'if [[ ! -z "$CKPID" ]]; then kill $CKPID; wait $CKPID; fi; if [[ ! -z "$LOGPID" ]]; then kill $LOGPID; wait $LOGPID; fi' EXIT; wait $CKPID; CKPID=""`

Let me know if it works or not. Although, this command won't save the world on server stop.
If it still doesn't work, then it's probably something on your end.

pgpais commented:
Hey, I was trying to setup a core keeper server and came across this just now.

I'm encountering the same issue as @ImKringle . So it might be something you have set up that we don't?

For context, I'm running a ubuntu server  24.04 as a VM in proxmox. I have been able to run other servers (satisfactory and valheim) so I don't think the issue lies on my side.

Please let me know if I can help in any way. We were just about to start playing this for the first time and I'd love to have a server for us :)

ImKringle commented:
Yeah this is happening on a bare metal install, this egg as well as just using a basic ubuntu or debian container. Its def just a missing package, question is *which one*

MasaruTobashi commented:
Hi, Same about me,

No starting with the eggs. I'm on Proxmox with a lxc for ptero. 

Tested with previous version and latest of the eggs. 

```
steam user is not set.
Using anonymous user.
WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
Redirecting stderr to '/home/container/Steam/logs/stderr.txt'
Logging directory: '/home/container/Steam/logs'
[  0%] Checking for available updates...
[----] Verifying installation...
UpdateUI: skip show logoSteam Console Client (c) Valve Corporation - version 1721172922
-- type 'quit' to exit --
Loading Steam API...OK
Connecting anonymously to Steam Public...OK
Waiting for client config...OK
Waiting for user info...OK
Success! App '1963720' already up to date.
 Update state (0x5) verifying install, progress: 14.72 (11534336 / 78382436)
Success! App '1007' fully installed.
:/home/container$ export DISPLAY=:0; rm ./GameID.txt ./CoreKeeperServerLog.txt; touch ./CoreKeeperServerLog.txt; xvfb-run -s "-screen 0 ${DISPLAY_WIDTH}x${DISPLAY_HEIGHT}x${DISPLAY_DEPTH} -ac -nolisten tcp -nolisten unix" ./CoreKeeperServer -batchmode -logfile CoreKeeperServerLog.txt -world ${WORLD_INDEX} -worldname "${WORLD_NAME}" -worldseed ${WORLD_SEED} $([[ "${GAME_ID}" != "" ]] && echo -n " -gameid ${GAME_ID}") -maxplayers ${MAX_PLAYERS} -worldmode ${WORLD_MODE} -port ${SERVER_PORT} & CKPID=$!; tail -f CoreKeeperServerLog.txt | grep -e '^\[userid:[0-9]*\] player' -e '^[^\[]' & LOGPID=$!; trap "kill ${CKPID}; wait ${CKPID}; kill ${LOGPID}; wait ${LOGPID}" 15; wait $!
rm: cannot remove './GameID.txt': No such file or directory
rm: cannot remove './CoreKeeperServerLog.txt': No such file or directory
[UnityMemory] Configuration Parameters - Can be set up in boot.config
    "memorysetup-bucket-allocator-granularity=16"
    "memorysetup-bucket-allocator-bucket-count=8"
    "memorysetup-bucket-allocator-block-size=4194304"
    "memorysetup-bucket-allocator-block-count=1"
    "memorysetup-main-allocator-block-size=16777216"
    "memorysetup-thread-allocator-block-size=16777216"
    "memorysetup-gfx-main-allocator-block-size=16777216"
    "memorysetup-gfx-thread-allocator-block-size=16777216"
    "memorysetup-cache-allocator-block-size=4194304"
    "memorysetup-typetree-allocator-block-size=2097152"
    "memorysetup-profiler-bucket-allocator-granularity=16"
    "memorysetup-profiler-bucket-allocator-bucket-count=8"
    "memorysetup-profiler-bucket-allocator-block-size=4194304"
    "memorysetup-profiler-bucket-allocator-block-count=1"
    "memorysetup-profiler-allocator-block-size=16777216"
    "memorysetup-profiler-editor-allocator-block-size=1048576"
    "memorysetup-temp-allocator-size-main=4194304"
    "memorysetup-job-temp-allocator-block-size=2097152"
    "memorysetup-job-temp-allocator-block-size-background=1048576"
    "memorysetup-job-temp-allocator-reduction-small-platforms=262144"
    "memorysetup-allocator-temp-initial-block-size-main=262144"
    "memorysetup-allocator-temp-initial-block-size-worker=262144"
    "memorysetup-temp-allocator-size-background-worker=32768"
    "memorysetup-temp-allocator-size-job-worker=262144"
    "memorysetup-temp-allocator-size-preload-manager=262144"
    "memorysetup-temp-allocator-size-nav-mesh-worker=65536"
    "memorysetup-temp-allocator-size-audio-worker=65536"
    "memorysetup-temp-allocator-size-cloud-worker=32768"
    "memorysetup-temp-allocator-size-gfx=262144"
Mono path[0] = '/home/container/CoreKeeperServer_Data/Managed'
Mono config path = '/home/container/CoreKeeperServer_Data/MonoBleedingEdge/etc'
Preloaded 'discord_game_sdk.so'
Shutdown handler: initialize.
Preloaded 'libEOSSDK-Linux-Shipping.so'
Preloaded 'libPlayFabMultiplayer.so'
Preloaded 'lib_burst_generated.so'
Caught fatal signal - signo:11 code:1 errno:0 addr:0x20
Obtained 24 stack frames.
#0  0x007fe9e935c050 in __sigaction
#1  0x007fe95ddf6ee0 in BumblelionLogger::TryToLoadConfigurationFile()
#2  0x007fe95ddf6cd8 in BumblelionLogger::BumblelionLogger()
#3  0x007fe95ddf7870 in BumblelionLogger::GetInstance()
#4  0x007fe95ddf6bf6 in DbgLogAreaFlags_FnInOut
#5  0x007fe95de0ed79 in OutstandingPlayFabRequestManager::OutstandingRequest::OutstandingRequest()
#6  0x007fe95de0ebfd in OutstandingPlayFabRequestManager::OutstandingRequestList::OutstandingRequestList()
#7  0x007fe95de146ab in _GLOBAL__sub_I_PlayFabServiceManagerImpl.cpp
#8  0x007fe9eb6959ce in __nptl_change_stack_perm
#9  0x007fe9eb695ab4 in __nptl_change_stack_perm
#10 0x007fe9e946f024 in _dl_catch_exception
#11 0x007fe9eb69c21e in _dl_rtld_di_serinfo
#12 0x007fe9e946efca in _dl_catch_exception
#13 0x007fe9eb69c5b8 in _dl_find_dso_for_object
#14 0x007fe9e93a5438 in dlmopen
#15 0x007fe9e946efca in _dl_catch_exception
#16 0x007fe9e946f07f in _dl_catch_error
#17 0x007fe9e93a4f27 in dlerror
#18 0x007fe9e93a54e9 in dlopen
#19 0x007fe9ea46df8e in PlayerMain(int, char**)
#20 0x007fe9ea45e77d in PlayerMain(int, char**)
#21 0x007fe9e934724a in __libc_init_first
#22 0x007fe9e9347305 in __libc_start_main
#23 0x00000000201029 in (Unknown)
Segmentation fault (core dumped)
```

QuintenQVD0 commented:
> Hi, Same about me,
> 
> No starting with the eggs. I'm on Proxmox with a lxc for ptero.
> 
> Tested with previous version and latest of the eggs.
> 
> ```
> steam user is not set.
> Using anonymous user.
> WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
> Redirecting stderr to '/home/container/Steam/logs/stderr.txt'
> Logging directory: '/home/container/Steam/logs'
> [  0%] Checking for available updates...
> [----] Verifying installation...
> UpdateUI: skip show logoSteam Console Client (c) Valve Corporation - version 1721172922
> -- type 'quit' to exit --
> Loading Steam API...OK
> Connecting anonymously to Steam Public...OK
> Waiting for client config...OK
> Waiting for user info...OK
> Success! App '1963720' already up to date.
>  Update state (0x5) verifying install, progress: 14.72 (11534336 / 78382436)
> Success! App '1007' fully installed.
> :/home/container$ export DISPLAY=:0; rm ./GameID.txt ./CoreKeeperServerLog.txt; touch ./CoreKeeperServerLog.txt; xvfb-run -s "-screen 0 ${DISPLAY_WIDTH}x${DISPLAY_HEIGHT}x${DISPLAY_DEPTH} -ac -nolisten tcp -nolisten unix" ./CoreKeeperServer -batchmode -logfile CoreKeeperServerLog.txt -world ${WORLD_INDEX} -worldname "${WORLD_NAME}" -worldseed ${WORLD_SEED} $([[ "${GAME_ID}" != "" ]] && echo -n " -gameid ${GAME_ID}") -maxplayers ${MAX_PLAYERS} -worldmode ${WORLD_MODE} -port ${SERVER_PORT} & CKPID=$!; tail -f CoreKeeperServerLog.txt | grep -e '^\[userid:[0-9]*\] player' -e '^[^\[]' & LOGPID=$!; trap "kill ${CKPID}; wait ${CKPID}; kill ${LOGPID}; wait ${LOGPID}" 15; wait $!
> rm: cannot remove './GameID.txt': No such file or directory
> rm: cannot remove './CoreKeeperServerLog.txt': No such file or directory
> [UnityMemory] Configuration Parameters - Can be set up in boot.config
>     "memorysetup-bucket-allocator-granularity=16"
>     "memorysetup-bucket-allocator-bucket-count=8"
>     "memorysetup-bucket-allocator-block-size=4194304"
>     "memorysetup-bucket-allocator-block-count=1"
>     "memorysetup-main-allocator-block-size=16777216"
>     "memorysetup-thread-allocator-block-size=16777216"
>     "memorysetup-gfx-main-allocator-block-size=16777216"
>     "memorysetup-gfx-thread-allocator-block-size=16777216"
>     "memorysetup-cache-allocator-block-size=4194304"
>     "memorysetup-typetree-allocator-block-size=2097152"
>     "memorysetup-profiler-bucket-allocator-granularity=16"
>     "memorysetup-profiler-bucket-allocator-bucket-count=8"
>     "memorysetup-profiler-bucket-allocator-block-size=4194304"
>     "memorysetup-profiler-bucket-allocator-block-count=1"
>     "memorysetup-profiler-allocator-block-size=16777216"
>     "memorysetup-profiler-editor-allocator-block-size=1048576"
>     "memorysetup-temp-allocator-size-main=4194304"
>     "memorysetup-job-temp-allocator-block-size=2097152"
>     "memorysetup-job-temp-allocator-block-size-background=1048576"
>     "memorysetup-job-temp-allocator-reduction-small-platforms=262144"
>     "memorysetup-allocator-temp-initial-block-size-main=262144"
>     "memorysetup-allocator-temp-initial-block-size-worker=262144"
>     "memorysetup-temp-allocator-size-background-worker=32768"
>     "memorysetup-temp-allocator-size-job-worker=262144"
>     "memorysetup-temp-allocator-size-preload-manager=262144"
>     "memorysetup-temp-allocator-size-nav-mesh-worker=65536"
>     "memorysetup-temp-allocator-size-audio-worker=65536"
>     "memorysetup-temp-allocator-size-cloud-worker=32768"
>     "memorysetup-temp-allocator-size-gfx=262144"
> Mono path[0] = '/home/container/CoreKeeperServer_Data/Managed'
> Mono config path = '/home/container/CoreKeeperServer_Data/MonoBleedingEdge/etc'
> Preloaded 'discord_game_sdk.so'
> Shutdown handler: initialize.
> Preloaded 'libEOSSDK-Linux-Shipping.so'
> Preloaded 'libPlayFabMultiplayer.so'
> Preloaded 'lib_burst_generated.so'
> Caught fatal signal - signo:11 code:1 errno:0 addr:0x20
> Obtained 24 stack frames.
> #0  0x007fe9e935c050 in __sigaction
> #1  0x007fe95ddf6ee0 in BumblelionLogger::TryToLoadConfigurationFile()
> #2  0x007fe95ddf6cd8 in BumblelionLogger::BumblelionLogger()
> #3  0x007fe95ddf7870 in BumblelionLogger::GetInstance()
> #4  0x007fe95ddf6bf6 in DbgLogAreaFlags_FnInOut
> #5  0x007fe95de0ed79 in OutstandingPlayFabRequestManager::OutstandingRequest::OutstandingRequest()
> #6  0x007fe95de0ebfd in OutstandingPlayFabRequestManager::OutstandingRequestList::OutstandingRequestList()
> #7  0x007fe95de146ab in _GLOBAL__sub_I_PlayFabServiceManagerImpl.cpp
> #8  0x007fe9eb6959ce in __nptl_change_stack_perm
> #9  0x007fe9eb695ab4 in __nptl_change_stack_perm
> #10 0x007fe9e946f024 in _dl_catch_exception
> #11 0x007fe9eb69c21e in _dl_rtld_di_serinfo
> #12 0x007fe9e946efca in _dl_catch_exception
> #13 0x007fe9eb69c5b8 in _dl_find_dso_for_object
> #14 0x007fe9e93a5438 in dlmopen
> #15 0x007fe9e946efca in _dl_catch_exception
> #16 0x007fe9e946f07f in _dl_catch_error
> #17 0x007fe9e93a4f27 in dlerror
> #18 0x007fe9e93a54e9 in dlopen
> #19 0x007fe9ea46df8e in PlayerMain(int, char**)
> #20 0x007fe9ea45e77d in PlayerMain(int, char**)
> #21 0x007fe9e934724a in __libc_init_first
> #22 0x007fe9e9347305 in __libc_start_main
> #23 0x00000000201029 in (Unknown)
> Segmentation fault (core dumped)
> ```

Lxc is not supported

HerrSammyDE commented:
I have the same problem on Debian 12.

I also found this article: https://discourse.cubecoders.com/t/core-keeper-segmentation-fault/3623

![image](https://github.com/user-attachments/assets/7c7b9503-81fb-45cd-84c5-f16714ae4912)

Osanchez commented:
for anyone with this issue. try removing/renaming the libparty.so file in /mnt/server/CoreKeeperServer_Data/Plugins/libparty.so

https://github.com/pelican-eggs/games-steamcmd/pull/70

bippity commented:
> for anyone with this issue. try removing/renaming the libparty.so file in /home/container/CoreKeeperServer_Data/Plugins/libparty.so
> 
> #70

Using the original/current repo's egg, I removed the file and was able to start the server and a gameID was outputted.
However, players were unable to actually connect to the server, they saw a "Connection Timed Out" error.

It wasn't until I replaced the default startup command with only `./_launch.sh` that the players were able to connect.
Not sure if that would cause any issues, but the server is now functional.

Osanchez commented:
@bippity the startup command in this egg is derived from the _launch.sh and it is advised that we do not call scripts in the launch file.
